analyses/focal-cn-file-preparation/02-add-ploidy-consensus.Rmd

```diff
@@ -0,0 +1,148 @@
+
+## Usage
+
+This notebook is intended to be run from the command line with the following (assumes you are in the root directory of the repository):
+
+```
+Rscript -e "rmarkdown::render('analyses/focal-cn-file-preparation/02-add-ploidy-consensus.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+library(tidyverse)
+```
+
+### Files and directories
+
+```{r}
+scratch_dir <- file.path("..", "..", "scratch")
+```
+
+```{r}
+# TODO: the consensus SEG file is not currently in the data download -- when it
+# gets included we will have to change the file path here
+consensus_seg_file <- file.path("..", "copy_number_consensus_call", "results", 
+                                "pbta-cnv-consensus.seg.gz")
+histologies_file <- file.path("..", "..", "data", "pbta-histologies.tsv")
+
+consensus_seg_df <- read_tsv(consensus_seg_file)
+histologies_df <- read_tsv(histologies_file,
+                           col_types = cols(
+                             molecular_subtype = col_character()
+                           ))
+```
+
+## Add ploidy and status
+
+As noted above, the sex chromosomes do not have copy number information included in the consensus SEG file, but other copy number values can be missing.
+
+From the [consensus SEG file methods](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/0e642ef0602ad04b8d0fbd80be626d4374fee928/analyses/copy_number_consensus_call#consensus-cnv-creation): 
+
+> The `copy.num` column is the weighted median of CNVkit segment values where they exist, or Control-FREEC values in the absence of CNVkit data. Because some software (notably GISTIC) requires all samples to have the same regions called, the copy number variants from `cnv_consensus.tsv` are supplementented with "neutral" segments where no call was made. These include all non-variant regions present in `ref/cnv_callable.bed` The neutral regions are assigned copy.num 2, except on chrX and chrY, where the copy number is left NA.
+
+For segments with missing `copy.num`, what chromosomes are they on?
+
+```{r}
+consensus_seg_df %>%
+  filter(is.na(copy.num)) %>%
+  group_by(chrom) %>%
+  tally()
+```
+
+The majority are on the sex chromosomes, which is expected.
+We can remove these, as we can not add status using the ploidy information without the `copy.num` information.
+We need to add in the `tumor_ploidy` column from the histologies file.
+
+```{r}
+add_ploidy_df <- consensus_seg_df %>%
+  filter(!is.na(copy.num)) %>%
+  inner_join(select(histologies_df, 
+                    Kids_First_Biospecimen_ID, 
+                    tumor_ploidy,
+                    germline_sex_estimate), 
+             by = c("ID" = "Kids_First_Biospecimen_ID")) %>%
+  select(-tumor_ploidy, -germline_sex_estimate, everything())
+```
+
+### Handle autosomes first
+
+```{r}
+add_autosomes_df <- add_ploidy_df %>%
+  # x and y chromosomes must be handled differently - leave in for safety
```

Comment:
leave in for safety?

Reply:
Ah, there was a moment where we thought the sex chromosomes were not represented in the consensus file at all. I was wrong and left in this comment.